fix: validate website URL in ContextJotaiActionsDiscovery

limichange opened the pull request:
- Added URL validation to ensure that the processed website URL is valid before further processing, enhancing the robustness of the website handling logic.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **Bug Fixes**
	- Enhanced website link processing by verifying addresses before use, ensuring smoother and more reliable navigation.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

limichange committed:
fix: validate website URL in ContextJotaiActionsDiscovery

- Added URL validation to ensure that the processed website URL is valid before further processing, enhancing the robustness of the website handling logic.

sidmorizon commented on packages/kit/src/states/jotai/contexts/discovery/actions.ts:
```diff
@@ -774,6 +774,7 @@ class ContextJotaiActionsDiscovery extends ContextJotaiActionsBase {
     ) => {
       if (webSite?.url) {
         webSite.url = processWebSiteUrl(webSite.url) ?? webSite.url;
```
uriUtils.validateUrl 放到 processWebSiteUrl 内部校验，否则其他人调用 processWebSiteUrl 还是可能出问题

limichange replied:
好的 👌

originalix committed:
Merge branch 'x' into OK-36708

limichange committed:
refactor: streamline website URL processing in discovery actions

- Removed redundant URL validation in ContextJotaiActionsDiscovery.
- Updated processWebSiteUrl function to utilize uriUtils for URL validation, enhancing clarity and maintainability of the URL handling logic.
